src/v/transform/logging/event.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "base/seastarx.h"
+#include "model/transform.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/sstring.hh>
+
+#pragma once
+
+namespace transform::logging {
+using transform_log_level
+  = named_type<ss::log_level, struct transform_log_level_tag>;
+
+uint32_t transform_log_level_to_severity(transform_log_level lvl);
+
+struct transform_log_event {
+    using clock_type = std::chrono::system_clock;
+    transform_log_event() = delete;
+    explicit transform_log_event(
+      model::node_id source,
+      clock_type::time_point ts,
+      ss::log_level level,
+      ss::sstring message)
+      : source_id(source)
```

Comment:
optional-nit: consider moving the constructor down to the implemenation file.

Reply:
done